FrontEnd/src/hooks/useAuth.js

```diff
@@ -1,6 +1,108 @@
 import { useContext } from 'react';
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import useSWR from 'swr';
+import axios from 'axios';
 import { AuthContext } from '../context';
 
-export default function useAuth() {
+export function AuthProvider ({ children }) {
+  const [isAuth, setIsAuth] = useState(!!JSON.parse(localStorage.getItem('isAuth')));
+  const [user, setUser] = useState(null);
+  const [isLoading, setLoading] = useState(true);
+  const authToken = localStorage.getItem('Token');
+  const navigate = useNavigate();
+
+  const { data, error, mutate } = useSWR(
+    authToken
+      ? [`${process.env.REACT_APP_BASE_API_URL}/api/auth/users/me/`, authToken]
+      : null,
+    ([url, authToken]) =>
+      fetch(url, {
+        method: 'GET',
+        headers: {
+          Authorization: `Token ${authToken}`,
+        },
+      })
+      .then((res) => {
```

Comment:
I think we should set isAuthTrue here when we validate our token. It could be expired

Reply:
In case the token has expired, user will be immediately logged out since any request returns 401 error. isAuth will be set to false and deleted from local storage.